fmriprep/workflows/confounds.py

```diff
@@ -0,0 +1,123 @@
+                        name='inputnode')
+    outputnode = pe.Node(utility.IdentityInterface(fields=['confounds_file']),
+                         name='outputnode')
+
+    # registration using ANTs
+    t1_registration = pe.Node(ants.ApplyTransforms(interpolation='MultiLabel'), name='TransformT1')
+    epi_registration = pe.Node(ants.WarpTimeSeriesImageMultiTransform(), name='TransformEPI')
+
+    # Global and segment regressors
+    signals = pe.Node(nilearn.SignalExtraction(include_global=True, detrend=True,
+                                               class_labels=FAST_DEFAULT_SEGS),
+                      name="SignalExtraction")
+    # DVARS
+    dvars = pe.Node(confounds.ComputeDVARS(save_all=True, remove_zerovariance=True),
+                    name="ComputeDVARS")
+    # Frame displacement
+    frame_displace = pe.Node(confounds.FramewiseDisplacement(), name="FramewiseDisplacement")
+    # CompCor
+    tcompcor = pe.Node(confounds.TCompCor(components_file='tcompcor.tsv'), name="tCompCor")
+    acompcor_roi = pe.Node(mask.BinarizeSegmentation(
+        false_values=[FAST_DEFAULT_SEGS.index('gray matter') + 1, 0]), # 0 denotes background
+                           name="CalcaCompCorROI")
+    acompcor = pe.Node(confounds.ACompCor(components_file='acompcor.tsv'), name="aCompCor")
+
+    # misc utilities
+    concat = pe.Node(utility.Function(function=_gather_confounds, input_names=['signals', 'dvars',
+                                                                               'frame_displace',
+                                                                               'tcompcor',
+                                                                               'acompcor'],
+                                      output_names=['combined_out']),
+                     name="ConcatConfounds")
+    ds_confounds = pe.Node(interfaces.DerivativesDataSink(base_directory=settings['output_dir'],
+                                                          suffix='confounds'),
+                           name="DerivConfounds")
+
+    workflow = pe.Workflow(name=name)
+    workflow.connect([
+        # connect inputnode to each non-anatomical confound node
+        (inputnode, dvars, [('fmri_file', 'in_file'),
+                            ('epi_mask', 'in_mask')]),
+        (inputnode, frame_displace, [('movpar_file', 'in_plots')]),
+        (inputnode, tcompcor, [('fmri_file', 'realigned_file')]),
+
+        # anatomically-based confound computation requires coregistration
+        (inputnode, t1_registration, [('reference_image', 'reference_image'),
+                                      ('t1_seg', 'input_image'),
+                                      ('t1_transform', 'transforms')]),
+        (inputnode, epi_registration, [('fmri_file', 'input_image'),
+                                       ('reference_image', 'reference_image'),
+                                       ('epi_transform', 'transformation_series')]),
```

Comment:
The `fmri_file` can be very big (up to 1.5GB per run) it would be much more efficient (in terms of working dir space and compute time) to move the t1_segmentation to the epi space without resampling or copying the `fmri_file`